app/vmselect/deployment/Dockerfile

```diff
@@ -1,8 +1,18 @@
 ARG certs_image
+ARG rootless_image
+
 FROM $certs_image AS certs
+FROM $rootless_image AS rootless
 FROM scratch
+
+COPY --from=rootless --chown=0:0 /etc/passwd /etc/passwd
+COPY --from=rootless --chown=0:0 /etc/group /etc/group
+USER 1000
+COPY --from=rootless --chown=1000 /home/victoriametrics /home/victoriametrics
 COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
+
 ARG src_binary
 COPY $src_binary ./vmselect-prod
+
 EXPOSE 8481
-ENTRYPOINT ["/vmselect-prod"]
+ENTRYPOINT ["/vmselect-prod", "--cacheDataPath=/home/victoriametrics"]
```

Comment:
This arg is not conflicting with `CMD/arg --cacheDataPath=` and may be overridden by it

Reply:
This may break existing setups after the upgrade if they didn't set `-cacheDataPath` arg.
Is there an approach to chown the `/` directory in the container, so the non-root user could write to it?

Reply:
If they didn't set `-cacheDataPath` it can have any default value which can change from image to image without breaking anything.
Probably you are talking about that cases, when `/tmp` is mounted as volume. In this case its not kind of "breaking experience", container just will ignore that previous cache)

I don't know ways to preserve "container runtime fs-layer" when upgrading its image, so this is not a case we're interested in.

I can't completely decide is it normal to chown whole `/` or even `/tmp`. Looks like it may be ok only in case when there is no other then VM software in image.

Looks like the only possible breaking change in this PR is changing of default `vmselect --cacheDataPath` and `vmstorage --storageDataPath` paths values.
We can choose which way is better:
1. Use `/home/victoriametrics` and assume possible volumes paths missmatch as edge case, or note it as a breaking change
1. `chown victoriametrics /tmp /vmstorage-data`
2. Experiment with `chown victoriametrics /` which means that only one user in container allowed to access something

Reply:
Let's try `chown victoriametrics /`. This won't break cases when users mount external volumes for default values of `-storageDatapPath` or similar flags like described at `https://hub.docker.com/r/victoriametrics/victoria-metrics`

Reply:
The main problem is that `/bin/chown` must be runned inside "end user image", not in intermediate multistage image.
(Examples below related to base-image strategy, which i focused on)
e.g.:
```Dockerfile
FROM alpine:3.10 as base
# ...
FROM alpine:3.10 # not a "scratch"
RUN chown 1000: /
# ...
```

I've tried some tricky way 
```Dockerfile
FROM alpine:3.10 as base
# ...
RUN adduser -u 1000 -S victoriametrics
RUN mkdir /future-root

FROM scratch
COPY --from=base --chown=1000: /future-root /
COPY --from=base --chown=1000:0 /etc/passwd /etc/passwd
COPY --from=base --chown=1000:0 /etc/group /etc/group
USER 1000
```
but got `mkdir /tmp: permission denied` 
<details>
  <summary>and it often fails with unknown error: (3 of 5 same builds fails on OS X)</summary>

```sh
$ docker image rm -f local/base:1.0.0 && make package-vmselect
Untagged: local/base:1.0.0
APP_NAME=vmselect /Library/Developer/CommandLineTools/usr/bin/make package-via-docker
(docker image ls --format '{{.Repository}}:{{.Tag}}' | grep -q 'docker.io/victoriametrics/vmselect:heads-rootless-0-g9f26f2a-dirty-120aa5c3$') || (\
		/Library/Developer/CommandLineTools/usr/bin/make app-via-docker && \
		docker build \
			--build-arg src_binary=vmselect-prod \
			--build-arg certs_image= \
			--build-arg base_image=local/base:1.0.0 \
			-t docker.io/victoriametrics/vmselect:heads-rootless-0-g9f26f2a-dirty-120aa5c3 \
			-f app/vmselect/deployment/Dockerfile bin)
(docker image ls --format '{{.Repository}}:{{.Tag}}' | grep -q 'local/base:1.0.0$') \
		|| docker build -t local/base:1.0.0 deployment/docker/base
Sending build context to Docker daemon  4.096kB
Step 1/8 : FROM alpine:3.10 as base
 ---> af341ccd2df8
Step 2/8 : RUN apk --update --no-cache add ca-certificates
 ---> Using cache
 ---> cb4637182acd
Step 3/8 : RUN mkdir /future-root
 ---> Running in 9fa334ed3ebe
Removing intermediate container 9fa334ed3ebe
 ---> a69558d7f01d
Step 4/8 : FROM scratch
 --->
Step 5/8 : USER 1000
 ---> Using cache
 ---> a7fbedf1a84b
Step 6/8 : COPY --from=base --chown=1000:1000 /future-root /
 ---> f3e2848c8cbb
Step 7/8 : COPY --from=base --chown=1000:0 /etc/passwd /etc/passwd
failed to export image: failed to create image: failed to get layer sha256:d63e3d3eff9f65202d3d4906441f3a64a122cfe35fb077cdbfc1e763aeee7897: layer does not exist
make[2]: *** [package-base] Error 1
make[1]: *** [package-via-docker] Error 2
make: *** [package-vmselect] Error 2
```

</details>

Replacing with
```diff
- RUN mkdir /future-root && chown -R victoriametrics: /future-root
+ RUN mkdir -p /future-root/tmp && chown -R victoriametrics: /future-root
```
works just fine.

Creating whole fs root for `chown`'ing `/tmp` seems pointless, so i've decided to use way №2 from previous comment

Reply:
I've updated the PR, so you can also look at base-image strategy implementation)